wcfsetup/install/files/lib/system/language/preload/command/CachePreloadPhrases.class.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace wcf\system\language\preload\command;
+
+use Laminas\Diactoros\Stream;
+use wcf\data\language\Language;
+use wcf\system\event\EventHandler;
+use wcf\system\language\preload\PreloadPhrasesCollecting;
+use wcf\util\StringUtil;
+
+/**
+ * Rebuilds the phrase preload cache for the
+ * requested language.
+ *
+ * @author Alexander Ebert
+ * @copyright 2001-2022 WoltLab GmbH
+ * @license GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @package WoltLabSuite\Core\System\Language\Preload\Command
+ * @since 6.0
+ */
+final class CachePreloadPhrases
+{
+    private readonly EventHandler $eventHandler;
+    private readonly Language $language;
+
+    public function __construct(Language $language)
+    {
+        $this->language = $language;
+
+        $this->eventHandler = EventHandler::getInstance();
+    }
+
+    public function __invoke(): void
+    {
+        $event = new PreloadPhrasesCollecting($this->language);
+        $this->eventHandler->fire($event);
+
+        $file = new Stream(\WCF_DIR . $this->language->getPreloadCacheFilename(), 'wb');
+        $file->write(
+            \sprintf(
+                "/* cache for '%s' (generated at %s) -- DO NOT EDIT */\n",
+                $this->language->getLocale(),
+                \gmdate('r'),
```

Comment:
Prefer creating a fresh `DateTimeImmutable` and the `c` format string, the functional API for date-time is not great.